### ISSUE TITLE
Create cr after delete same cr, it will error.

### ISSUE THREAD

**Issue by Mrart** [GoogleCloudPlatform/flink-on-k8s-operator#249]
When delete cr a, and create a . cr a will get the terminating job pods， then the cr status is failed when create the cr a.

**Comment by functicons** [comment]
@Mrart I tried but couldn't reproduce the problem with the sample CRs included in the project. Could you give your steps and commands for reproducing?

**Comment by Mrart** [comment]
@functicons  when the job pods is “ Terminating”，we creating the flinkcluster, it will reproducing.

**Comment by functicons** [comment]
This sounds like a race condition. We might need to wait until resources of the previous cluster with the same name are cleaned up before creating resources for the new one.

**Comment by functicons** [comment]
@Mrart do you mind adding your name to this [doc](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/docs/who-is-using.md)?

**Comment by Mrart** [comment]
@functicons  Can we fixed this only waiting  delete job pods finished ?

**Comment by functicons** [comment]
I think the wait has to be after the new CR is created, because when deleting the CR, the operator has no control, the child resources are auto deleted.

**Comment by Mrart** [comment]
I agree 'the wait has to be after the new CR is created,'  The operator has used  ‘name’ for distinguish the flinkcluster 。

**Comment by Mrart** [comment]
@functicons  I think, We can use OwnerReference#BlockOwnerDeletion=true for delete flinkclusters job pods. When delete flinkclusters, the job pods must be deleted, to confirm the pods has be delete before the flinkclusters has been deleted.
